ui/src/common/actions.ts

```diff
@@ -143,6 +160,88 @@ function removeTrack(state: StateDraft, trackId: string) {
   state.pinnedTracks = state.pinnedTracks.filter((id) => id !== trackId);
 }
 
+// A helper to clean the state for a given removable track group.
+function removeTrackGroup(state: StateDraft, groupId: string) {
+  delete state.trackGroups[groupId];
+
+  state.pinnedTracks = state.pinnedTracks.filter((id) => id !== groupId);
+}
+
+function hasRemovable<T extends AddTrackLikeArgs>(item: T): item is T & {isRemovable: boolean} {
+  return 'isRemovable' in item && typeof item.isRemovable === 'boolean';
+}
+
+function wasFiltered<T extends TrackState|TrackGroupState>(item: T): item is T & {wasFiltered?: true} {
+  return 'wasFiltered' in item && item.wasFiltered === true;
+}
+
+function isFilteredTrack(track: Partial<AddTrackArgs>): boolean {
+  return globals.trackFilteringEnabled && globals.filteredTracks
+      .some(filtered => isAddTrackArgs(filtered) && isSameTrack(filtered, track));
+}
+
+// Query whether an |other| track matches enough details of a |track| as
+// to represent the same track
+function isSameTrack(track: AddTrackArgs, other: Partial<AddTrackArgs>): boolean {
+  return track.kind === other.kind
+      && track.trackGroup == other.trackGroup
+      // TODO: This may not be reliable. May need to deep-compare the config object
```

Comment:
Should we create a follow-up task for this? To make sure this comment is either removed or resolved?